### PR TITLE
Querytracker tests

### LINKED ISSUE #376: Add units tests in the querytracker pkg
Add unit tests to cover the missed functions in querytracker.go file.

(Optional) Testing
Make sure all the tests added pass.

Acceptance Criteria
These tests should increase the code coverage for the querytracker package.

Current Coverage Report:
image

### PR DESCRIPTION
**Pull request by SergeyPishchulov**
# Description
Add tests. Add error messages. Decomposed one function.

Fixes #376 

# Testing
Tests are in querytracker_test.go

# Checklist:
Before marking your pull request as ready for review, complete the following.

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

**Comment by SergeyPishchulov** [comment]
Hello!

I have added some tests as was mentioned in the issue. I would be grateful if you review the changes. 

I plan to add some tests for the remaining functions in next PR. Thank you 🙏

File: pkg/querytracker/querytracker.go [hunk 1]
```diff
@@ -945,6 +956,7 @@ func parsePostPqsAggBody(jsonSource map[string]interface{}) error {
 	return nil
 }
 func processPostAggs(inputValueParam interface{}) (map[string]bool, error) {
+	// asserts that inputValueParam is slise of strings
```
Comment (AndrewHess) [inline pkg/querytracker/querytracker.go:959]: typo: 
```suggestion
	// asserts that inputValueParam is a slice of strings
```

File: pkg/querytracker/querytracker_test.go [hunk 2]
```diff
@@ -385,3 +386,106 @@ func Test_PostPqsClear(t *testing.T) {
...
+}
+
+func Test_processPostAggs_NoErrorWhenSliceOfStrings(t *testing.T) {
+	key := "new_column"
+	inputValueParam := []interface{}{key}
+	got, err := processPostAggs(inputValueParam)
+	assert.Nil(t, err)
+	assert.True(t, got[key])
+}
+func Test_processPostAggs_ErrorWhenSliceOfNotStrings(t *testing.T) {
```
Comment (AndrewHess) [inline pkg/querytracker/querytracker_test.go:430]: nit: add a blank line between the functions

Same comment for lines 464-465

File: pkg/querytracker/querytracker_test.go [hunk 3]
```diff
@@ -385,3 +386,106 @@ func Test_PostPqsClear(t *testing.T) {
...
+	assert.Nil(t, err)
+}
+
+func Test_processPostAggs_NoErrorWhenSliceOfStrings(t *testing.T) {
+	key := "new_column"
+	inputValueParam := []interface{}{key}
+	got, err := processPostAggs(inputValueParam)
+	assert.Nil(t, err)
+	assert.True(t, got[key])
+}
```
Comment (AndrewHess) [inline pkg/querytracker/querytracker_test.go:429]: Also check the length of the map

**Review by AndrewHess — COMMENTED** [review]
This looks good. I just have some minor comments

File: pkg/querytracker/querytracker.go [hunk 4]
```diff
@@ -925,11 +930,17 @@ func parsePostPqsAggBody(jsonSource map[string]interface{}) error {
 				}
 			}
 		default:
-			log.Errorf("PostPqsAggCols: Invalid key=[%v]", key)
-			err := fmt.Sprintf("PostPqsAggCols: Invalid key=[%v]", key)
+			message := fmt.Sprintf("PostPqsAggCols: Invalid key=[%v] with value of type [%v]", key, reflect.TypeOf(value))
```
[the remark this answers is not in the document]
Comment (SergeyPishchulov) [reply pkg/querytracker/querytracker.go:933]: Done

File: pkg/querytracker/querytracker.go [hunk 5]
```diff
@@ -925,11 +930,17 @@ func parsePostPqsAggBody(jsonSource map[string]interface{}) error {
...
+			log.Error(message)
+			err := fmt.Sprint(message)
 			return errors.New(err)
```
[the remark this answers is not in the document]
Comment (SergeyPishchulov) [reply pkg/querytracker/querytracker.go:936]: Done

**Comment by SergeyPishchulov** [comment]
Thank you for the review.
Fixed all mistakes mentioned in the comments. 🙂

**Review by AndrewHess — APPROVED** [review]
Looks good!

**Pull request merged**